std/algorithm/sorting.d

```diff
@@ -130,55 +132,89 @@ unittest
     assert(b == [ 3, 4, 5, 6 ]);
 }
 
-// isSorted
-/**
-Checks whether a forward range is sorted according to the comparison
-operation $(D less). Performs $(BIGOH r.length) evaluations of $(D
-less).
-
-Params:
-    less = Predicate the range should be sorted by.
-    r = Forward range to check for sortedness.
-
-Returns: true if the range is sorted, false otherwise.
-*/
-bool isSorted(alias less = "a < b", Range)(Range r) if (isForwardRange!(Range))
+private bool isSortedImpl(alias less = "a < b", bool isStrict = false, Range)(Range r)
+if (isForwardRange!(Range))
 {
     if (r.empty) return true;
 
     static if (isRandomAccessRange!Range && hasLength!Range)
     {
-        immutable limit = r.length - 1;
+        immutable limit = r.length > 0 ? r.length - 1 : 0;
         foreach (i; 0 .. limit)
         {
-            if (!binaryFun!less(r[i + 1], r[i])) continue;
-            assert(
-                !binaryFun!less(r[i], r[i + 1]),
-                "Predicate for isSorted is not antisymmetric. Both" ~
-                        " pred(a, b) and pred(b, a) are true for certain values.");
-            return false;
+            static if(!isStrict)
```

Comment:
Nitpick, can be `static if (isStrict) { … } else { … }` by switching the order of the clauses. Also, please put a space between `if` and opening parenthesis